aiida/cmdline/commands/cmd_graph.py

```diff
@@ -37,33 +48,49 @@ def verdi_graph():
 @click.option(
     '-d',
     '--descendant-depth',
-    help='The maximum depth when recursing through the descendants, if not set it will recurse to the end',
+    help='The maximum depth when recursing through the descendants. If not set it will recurse to the end.',
     type=click.IntRange(min=0))
-@click.option('-o', '--outputs', is_flag=True, help='Always show all outputs of a calculation.')
-@click.option('-i', '--inputs', is_flag=True, help='Always show all inputs of a calculation.')
+@click.option('-o', '--process-out', is_flag=True, help='Show outgoing links for all processes.')
+@click.option('-i', '--process-in', is_flag=True, help='Show incoming links for all processes.')
+@options.VERBOSE(help="Print verbose information of the graph traversal.")
 @click.option(
-    '-f',
-    '--output-format',
-    help='The output format, something that can be recognized by graphviz'
-    '(see http://www.graphviz.org/doc/info/output.html).',
+    '-e',
+    '--engine',
+    help="The graphviz engine, e.g. 'dot', 'circo', ... "
+    "(see http://www.graphviz.org/doc/info/output.html)",
     default='dot')
+@click.option('-f', '--output-format', help="The output format used for rendering ('pdf', 'png', etc.).", default='pdf')
+@click.option('-s', '--show', is_flag=True, help="Open the rendered result with the default application.")
 @decorators.with_dbenv()
-def generate(root_node, ancestor_depth, descendant_depth, outputs, inputs, output_format):
-    """Generate a graph for a given ROOT_NODE."""
-    from aiida.tools.visualization.graphviz import draw_graph
+def generate(root_node, link_types, id_label, ancestor_depth, descendant_depth, process_out, process_in, engine,
+             verbose, output_format, show):
+    """
+    Generate a graph from a ROOT_NODE (specified by pk or uuid).
```

Comment:
I think technically the node label could also be used. Granted it won't be that common. I think it would maybe be better to use the term "identifier". This is used across the CLI to mean one of PK, UUID, or label.

Reply:
fixed in 6e599e9